### PR TITLE
here command

### PR DESCRIPTION
**Pull request by brenfan**
useful for seeing who has access to a channel

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
 # setting up path hierarchy for commands to load
 commandFolder="commands_to_load."
-the_commands=[commandFolder+"HealthChecks", commandFolder+"Misc", commandFolder+"RoleCommands", commandFolder+"Administration"]
+the_commands=[commandFolder+"HealthChecks", commandFolder+"Misc", commandFolder+"RoleCommands", commandFolder+"Administration", commandFolder+"Here"]
```
Comment (modernNeo) [inline main.py]: this version of doing things is outdated, look as master for how VJ set it up

**Comment by TitanVJ** [comment]
update ur branch then go from there. cog loading has been changed as jace said.

File: commands_to_load/Here.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+
+class Here():
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    def build_embed(members, channel):
+        # build response
+        embed = discord.Embed(type = "rich")
```
Comment (modernNeo) [inline commands_to_load/Here.py]: please use the embed stuff that VJ made in the `helper_files/embed.py`

File: commands_to_load/Here.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            for member in members:
+                display = member.display_name + "   (" + str(member) + ")\n"
+                string += display
+
+        embed.description = string
+        return embed
+
+    @commands.command()
+    async def here(self, ctx, *search):
+        logger.info("[HereCommands here()] "
```
Comment (modernNeo) [inline commands_to_load/Here.py]: should start as `logger.info("[Here here()] "` to match convention.

also please add some more logs, one doesnt really seem like enough.

File: commands_to_load/Here.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    def __init__(self, bot):
+        self.bot = bot
+
+    def build_embed(members, channel):
+        # build response
+        embed = discord.Embed(type = "rich")
+        embed.title = "Users in **" + channel.name + "**"
+        embed.color = discord.Color.blurple()
+        embed.set_footer(text="brenfan", icon_url="https://i.imgur.com/vlpCuu2.jpg")
+        if len(members) == 0:
```
Comment (modernNeo) [inline commands_to_load/Here.py]: please include some logging to indicate

1. How many members there are

1. What the title will be set to

**Comment by modernNeo** [comment]
also need to add it to the help.json file along with an explanation of what the command is, how to call it and what the optional argument does

File: commands_to_load/Here.py (hunk 4 above)
Comment (brenfan) [reply commands_to_load/Here.py]: done

**Comment by modernNeo** [comment]
please checkout commands_to_load/help.json correctly and then add `.here` to it. VJ inadvertently did that wrong because the PR is currently overwriting help.json which isnt the correct way to do it.
`git checkout here`
`git remote add csss_origin https://github.com/CSSS/wall_e.git`
`git fetch csss_origin`
`git checkout csss_origin/master -- commands_to_load/help.json`
and then add `.here` to there and do a commit to your here branch. that should fix the issue.

**Comment by modernNeo** [comment]
I will look into why this doesn't want to work after tutorial

**Comment by brenfan** [comment]
zzz i fiex it

**Comment by modernNeo** [comment]
@brenfan good job on clearing the help.json. However, I want something more descriptive than "arg is used for filtering". you can either provide an example of how it is used or describe it in more detail on what is being filtered.

you also dont seem to have added it to the README.md

**Comment by modernNeo** [comment]
OK, please give an example of what an arg would be and then I will approve this PR.

**Comment by brenfan** [comment]
modernNeo

File: README.md [hunk 5]
```diff
@@ -24,6 +24,7 @@ Wall-E, named after the lovable character [Wall-E](https://en.wikipedia.org/wiki
 * `.iamn <arg>` - removes you from role `<arg>`
 * `.whois <arg>` - returns everyone who has role `<arg>`
 * `.roles` - displays all roles that exist on the server
+* `.here <args>` - displays all users with permissions to view the current channel. Filters results if they do not match any args, or all users if no args are given.
```
Comment (modernNeo) [inline README.md]: change it to 
* `.here [<arg>]` - displays all users with permissions to view the current channel. Results can be filter by looking for users whose username or nickname on the server contains the substring indicated with any of the included `<arg>` strings or all users if no args are given. Multiple `<args>` may be entered.

File: commands_to_load/help.json [hunk 6]
```diff
@@ -27,10 +27,11 @@
...
+		{ "access": "Bot_manager", "name": ".exc <arg>", "description": "executes the command <arg> on the bot server. The classes and their corresponding commands can be determined using the \".help\" command."},
 
 		{ "access": "Bot_manager", "name": "Class: Minions"},
 		{ "access": "Minions", "name": ".embed <arg0> <arg1> ...", "description": "Allows Minions to post embed messages.\nFor odd number of arguments the first arg will be used as description in the embed and the rest as field title and content.\nFor even number there will be no description."},
-		{ "access": "Minions", "name": ".modspeak <arg>", "description": "Posts the warning message <arg> in embed format. No quotes required with message."}
+		{ "access": "Minions", "name": ".modspeak <arg>", "description": "Posts the warning message <arg> in embed format. No quotes required with message."},
+		{ "access": "public", "name": ".here <args> ...", "description": "Tells you who's here. Filters results if they do not match any args, or all users if no args are given."}
```
Comment (modernNeo) [inline commands_to_load/help.json]: change description to

> Displays all users with permissions to view the current channel. Results can be filter by looking for users whose username or nickname on the server contains the substring indicated with any of the included <arg> strings or all users if no args are given. Multiple <args> may be entered.

Comment (brenfan) [reply commands_to_load/help.json]: its not that complicated but ok...

**Comment by brenfan** [comment]
i cant merge it

**Pull request merged**